tidalsim/bb/elf.py

```diff
@@ -104,42 +103,29 @@ def get_target_from_control_instr(match: Match) -> Tuple[Optional[int], Optional
 
     while func_start_pc := find_next_func(f):
         func_end_pc = parse_func(f)
-        basic_blocks.add(Interval(func_start_pc, func_end_pc, next_bbid()))
+        intervals += [(func_start_pc, func_end_pc)]
 
     logging.info(f"Found {len(all_control_instrs)} control instructions")
     logging.info(f"No target was identified for {no_target_identified} instructions")
 
-    return all_control_instrs, basic_blocks
-
-def get_split_bbid(iv: Interval, islower: bool) -> int:
-    if islower:
-        return iv.data
-    else:
-        return next_bbid()
+    return all_control_instrs, intervals
 
 # Splitting logic
-def do_basic_block_analysis(all_control_instrs: List[ObjdumpInstrEntry], initial_basic_blocks: IntervalTree) -> IntervalTree:
-
-    all_basic_blocks = initial_basic_blocks
-
-    def start_block_at(pc: int) -> None:
-        all_basic_blocks.slice(pc, get_split_bbid)
-
-    def end_block_after(control_instr: ObjdumpInstrEntry) -> None:
-        all_basic_blocks.slice(get_next_pc(control_instr), get_split_bbid)
+def do_basic_block_analysis(all_control_instrs: List[ObjdumpInstrEntry], initial_intervals: List[Interval]) -> List[Marker]:
+    events = intervals_to_events(initial_intervals)
 
     for control_instr in tqdm(all_control_instrs):
         if control_instr.target is None:
-            end_block_after(control_instr)
+            events += [cast(Event, (control_instr.pc, None))]
```

Comment:
Can we avoid using cast if we annotate the type of `events`?

Reply:
Not a pythonista though, so take with a grain of salt 🧂.

Reply:
Yeah I think type aliases are a bit poorly handled, this is all good